android/src/main/java/com/reactnativereceivesharingintent/ReceiveSharingIntentHelper.java

```diff
@@ -22,164 +32,439 @@ public class ReceiveSharingIntentHelper {
+      file.putString("contentUri", contentUri.toString());
+      file.putString("text", null);
       file.putString("weblink", null);
       file.putString("subject", subject);
-      files.putMap("0",file);
-    }else if(Objects.equals(intent.getAction(), Intent.ACTION_SEND_MULTIPLE)) {
-      ArrayList<Uri> contentUris = intent.getParcelableArrayListExtra(Intent.EXTRA_STREAM);
+
+      WritableMap exif = getExif(contentUri.toString());
+      file.putMap("exif", exif);
+
+      files.putMap("0", file);
+    } else if (
+      Objects.equals(intent.getAction(), Intent.ACTION_SEND_MULTIPLE)
+    ) {
+      ArrayList<Uri> contentUris = intent.getParcelableArrayListExtra(
+        Intent.EXTRA_STREAM
+      );
       if (contentUris != null) {
         int index = 0;
         for (Uri uri : contentUris) {
           WritableMap file = new WritableNativeMap();
           ContentResolver contentResolver = context.getContentResolver();
-          String filePath = FileDirectory.INSTANCE.getAbsolutePath(context, uri);
+          String filePath = FileDirectory.INSTANCE.getAbsolutePath(
+            context,
+            uri
+          );
+
           // Based on https://developer.android.com/training/secure-file-sharing/retrieve-info
           file.putString("mimeType", contentResolver.getType(uri));
-          Cursor queryResult = contentResolver.query(uri, null, null, null, null);
+          Cursor queryResult = contentResolver.query(
+            uri,
+            null,
+            null,
+            null,
+            null
+          );
           queryResult.moveToFirst();
-          file.putString("fileName", queryResult.getString(queryResult.getColumnIndex(OpenableColumns.DISPLAY_NAME)));
+          file.putString(
+            "fileName",
+            queryResult.getString(
+              queryResult.getColumnIndex(OpenableColumns.DISPLAY_NAME)
+            )
+          );
           file.putString("filePath", filePath);
-          file.putString("contentUri",uri.toString());
-          file.putString("text",null);
+          file.putString("contentUri", uri.toString());
+          file.putString("text", null);
           file.putString("weblink", null);
           file.putString("subject", subject);
-          files.putMap(Integer.toString(index),file);
+
+          WritableMap exif = getExif(uri.toString());
+          files.putMap("exif", exif);
+
+          files.putMap(Integer.toString(index), file);
+
           index++;
         }
       }
     }
-    return  files;
+    return files;
+  }
+
+  public WritableMap getExif(String uri) {
+    try {
+      ExifInterface exif = createExifInterface(uri);
+
+      WritableMap exifMap = new WritableNativeMap();
+
+      for (String attribute : EXIF_ATTRIBUTES) {
+        String value = exif.getAttribute(attribute);
+
+        // Rename props for convenience
+        if (attribute == ExifInterface.TAG_IMAGE_LENGTH) {
+          exifMap.putString("height", value);
```

Comment:
Have you tested the case when image is rotated?
Here is the article about it => https://sirv.com/help/articles/rotate-photos-to-be-upright/

In app we have helper function (not sure if it's correct though)
`src/Shop/util/conversion.ts`

Reply:
I looked into this, and the exif data returns orientation const which matched the logic in shop utils. I also added [MediaStore.Images.Media.ORIENTATION](https://developer.android.com/reference/android/media/MediaMetadataRetriever#METADATA_KEY_IMAGE_ROTATION) prop as a fallback which returns rotated degrees in case we want to add/change some logic later.

However, I found some issues on my Samsung device - exif orientation prop isn't returned correctly from the native side (eg. portrait photo rotated to landscape returns 0 as if it was not rotated, and original portrait photo shows 1 as if it is rotated)

I found a few similar issues on StackOverflow, but couldn't find any proper solution, so I'm not sure what to do about that, it could be Samsung-only issue, but I'm not sure :/

Reply:
It definitely device related issue, last time i researched such issue was device bug
Maybe there is a setting on your Samsung for that.

Note: if you need such helper as in shop please copy it to not depend on Shop part

If that takes too much time please raise a ticket with all your findings regarding this usecase

Reply:
Also the problem can be in exifVersion that device supports, at least with capture time there is an issue

Reply:
I looked into this a bit more, but wasn't able to find a solution to get proper orientation from native side. :( 
I opened a tech ticket so that it can be worked on in the future

Reply:
Thank you for the efforts